src/hardware/sblaster.cpp

```diff
@@ -294,6 +308,157 @@ static void InitializeSpeakerState()
 	}
 }
 
+static void configure_filters(Section_prop* config)
+{
+	auto get_filter_params = [&](const char * conf) {
+		static const std::map<SB_TYPES, FilterType> sb_type_to_filter_type_map = {
+				{SBT_NONE, FilterType::None},
+				{SBT_1, FilterType::SB1},
+				{SBT_2, FilterType::SB2},
+				{SBT_PRO1, FilterType::SBPro1},
+				{SBT_PRO2, FilterType::SBPro2},
+				{SBT_16, FilterType::SB16},
+				{SBT_GB, FilterType::None},
+		};
+
+		static const std::map<std::string, FilterType> filter_map = {
+				{"none", FilterType::None},
+				{"sb1", FilterType::SB1},
+				{"sb2", FilterType::SB2},
+				{"sbpro1", FilterType::SBPro1},
+				{"sbpro2", FilterType::SBPro2},
+				{"sb16", FilterType::SB16},
+		};
+		assert(conf);
+		const auto filter_prefs = split(config->Get_string(conf));
+		const auto filter = filter_prefs.empty() ? "auto" : filter_prefs[0];
+
+		FilterType filter_type = FilterType::None;
+		if (filter == "auto") {
+			auto it = sb_type_to_filter_type_map.find(sb.type);
+			if (it != sb_type_to_filter_type_map.end())
+				filter_type = it->second;
+		} else {
+			auto it = filter_map.find(filter);
+			if (it != filter_map.end())
+				filter_type = it->second;
+		}
+
+		FilterState filter_state = FilterState::Off;
+		if (filter_type != FilterType::None) {
+			if (filter_prefs.size() > 1 && filter_prefs[1] == "always_on") {
+				filter_state = FilterState::ForcedOn;
+			} else {
+				filter_state = FilterState::On;
+			}
+		}
+		return std::make_tuple(filter_type, filter_state);
+	};
+	using std::tie;
+	tie(sb.sb_filter_type, sb.sb_filter_state) = get_filter_params("sb_filter");
+	tie(sb.opl_filter_type, std::ignore) = get_filter_params("opl_filter");
+}
+
+static void log_filter_config(const char *output_type, FilterType filter)
+{
+	static const std::map<FilterType, std::string> filter_name_map = {
+	        {FilterType::SB1, "Sound Blaster 1.0"},
+	        {FilterType::SB2, "Sound Blaster 2.0"},
+	        {FilterType::SBPro1, "Sound Blaster Pro 1"},
+	        {FilterType::SBPro2, "Sound Blaster Pro 2"},
+	        {FilterType::SB16, "Sound Blaster 16"},
+	};
+
+	if (filter == FilterType::None) {
+		LOG_MSG("%s: %s filter emulation disabled", CardType(), output_type);
+	} else {
+		auto it = filter_name_map.find(filter);
+		if (it != filter_name_map.end()) {
+			auto filter_type = it->second;
+			LOG_MSG("%s: Emulating %s %s output filter",
```

Comment:
Can this LOG be reworded without "`Emulating ..."`? Just avoiding some redundancy, as we could in theory prefix most status lines with `"Emulating ..`.

``` text
2022-05-28 09:57:40.903 | SBPRO2: Emulating Sound Blaster Pro 2 OPL output filter
2022-05-28 09:57:40.904 | MIXER: SB channel operating at 22050 Hz and upsampling to the output rate
2022-05-28 09:57:40.904 | MIXER: SB channel filter enabled
2022-05-28 09:57:40.904 | SBPRO2: Emulating Sound Blaster Pro 2 PCM output filter
2022-05-28 09:57:40.904 | SBPRO2: Running on port 220h, irq=7, dma8=1, dma16=5
2022-05-28 09:57:40.904 | SBPRO2: @SET BLASTER=A220 I7 D1 T4
2022-05-28 09:57:40.904 | MIXER: SPKR channel operating at 18939 Hz and upsampling to the output rate
2022-05-28 09:57:40.904 | MIXER: TANDY channel operating at 48000 Hz without resampling
2022-05-28 09:57:40.908 | TANDY: Initialized audio card with a TI NCR 8496 PSG but no DAC, because a Sound Blaster is present
2022-05-28 09:57:40.908 | MIXER: DISNEY channel operating at 10000 Hz and upsampling to the output rate
```

Reply:
Sure, I did reword it a few times until I've arrived this but wasn't 100% happy. Changed one more time now 😂 

Btw, I had the same line of thought about using the word "emulate" first, but then checked the config descriptions and it's used in a lot of places (20+). So I figured it's kinda okay.

Reply:
> Btw, I had the same line of thought about using the word "emulate" first, but then checked the config descriptions and it's used in a lot of places (20+). So I figured it's kinda okay.

ahh ok! New message looks good.